src/Workspaces/Core/Portable/Workspace/Solution/SolutionState.CompilationTracker.State.cs

```diff
@@ -169,8 +181,7 @@ private sealed class InProgressState : State
                     : base(compilationWithoutGeneratedDocuments: new ConstantValueSource<Optional<Compilation>>(inProgressCompilation),
                            declarationOnlyCompilation: null,
                            generatedDocuments,
-                           generatedDocumentsAreFinal: false, // since we have a set of transformations to make, we'll always have to run generators again
-                           GetUnrootedSymbols(inProgressCompilation))
```

Comment:
Was there a reason we did this versus just passing null for the unrooted symbols in the first place? Because I would have _imagined_ the only times we ever handed out a compilation was after it had been passed to a FinalState. In the case of a compilation being made for frozen partial semantics, we still create a FinalState with the in-progress compilation before handing it out, so I would have _thought_ that would work.

The only place I can potentially imagine that invariant being broken is in GetPartialMetadataReference where it looks like we're just more or less returning a compilation directly, and it looks like there's already multiple _other_ bugs there that we may need to fix.